### PR TITLE
Remove `in_band_lifetimes` from `rustc_typeck`

### PR DESCRIPTION
**Pull request by Patrick-Poitras**
Joining in on the effort to remove the `in_band_lifetimes` features, as described in issue #91867.

### CODE REVIEW

File: compiler/rustc_typeck/src/check/wfcheck.rs [hunk 1]
```diff
@@ -1696,13 +1696,13 @@ pub struct CheckTypeWellFormedVisitor<'tcx> {
     tcx: TyCtxt<'tcx>,
 }
 
-impl CheckTypeWellFormedVisitor<'tcx> {
-    pub fn new(tcx: TyCtxt<'tcx>) -> CheckTypeWellFormedVisitor<'tcx> {
+impl CheckTypeWellFormedVisitor<'_> {
+    pub fn new(tcx: TyCtxt<'_>) -> CheckTypeWellFormedVisitor<'_> {
```
Comment (jackh726) [inline compiler/rustc_typeck/src/check/wfcheck.rs:1700]: I don't think this change is better

File: compiler/rustc_typeck/src/check_unused.rs [hunk 2]
```diff
@@ -21,7 +21,7 @@ pub fn check_crate(tcx: TyCtxt<'_>) {
     unused_crates_lint(tcx);
 }
 
-impl ItemLikeVisitor<'v> for CheckVisitor<'tcx> {
+impl <'v, 'tcx> ItemLikeVisitor<'v> for CheckVisitor<'tcx> {
```
Comment (jackh726) [inline compiler/rustc_typeck/src/check_unused.rs:24]: Can probably do `ItemLikeVisitor<'_>` instead

File: compiler/rustc_typeck/src/coherence/builtin.rs [hunk 3]
```diff
@@ -108,7 +108,7 @@ fn visit_implementation_of_copy(tcx: TyCtxt<'_>, impl_did: LocalDefId) {
     }
 }
 
-fn visit_implementation_of_coerce_unsized(tcx: TyCtxt<'tcx>, impl_did: LocalDefId) {
+fn visit_implementation_of_coerce_unsized(tcx: TyCtxt<'_>, impl_did: LocalDefId) {
```
Comment (jackh726) [inline compiler/rustc_typeck/src/coherence/builtin.rs:111]: Maybe name `'tcx` here

File: compiler/rustc_typeck/src/coherence/inherent_impls.rs [hunk 4]
```diff
@@ -38,7 +38,7 @@ struct InherentCollect<'tcx> {
     impls_map: CrateInherentImpls,
 }
 
-impl ItemLikeVisitor<'v> for InherentCollect<'tcx> {
+impl<'v, 'tcx> ItemLikeVisitor<'v> for InherentCollect<'tcx> {
```
Comment (jackh726) [inline compiler/rustc_typeck/src/coherence/inherent_impls.rs:41]: Can probably elide `'v` here

File: compiler/rustc_typeck/src/coherence/inherent_impls_overlap.rs [hunk 5]
```diff
@@ -115,7 +115,7 @@ impl InherentOverlapChecker<'tcx> {
     }
 }
 
-impl ItemLikeVisitor<'v> for InherentOverlapChecker<'tcx> {
+impl<'v, 'tcx> ItemLikeVisitor<'v> for InherentOverlapChecker<'tcx> {
```
Comment (jackh726) [inline compiler/rustc_typeck/src/coherence/inherent_impls_overlap.rs:118]: Also here

File: compiler/rustc_typeck/src/coherence/unsafety.rs [hunk 6]
```diff
@@ -16,8 +16,8 @@ struct UnsafetyChecker<'tcx> {
     tcx: TyCtxt<'tcx>,
 }
 
-impl UnsafetyChecker<'tcx> {
-    fn check_unsafety_coherence(
+impl<'tcx> UnsafetyChecker<'tcx> {
+    fn check_unsafety_coherence<'v>(
         &mut self,
         item: &'v hir::Item<'v>,
```
Comment (jackh726) [inline compiler/rustc_typeck/src/coherence/unsafety.rs:22]: Here too

File: compiler/rustc_typeck/src/coherence/unsafety.rs [hunk 7]
```diff
@@ -83,7 +83,7 @@ impl UnsafetyChecker<'tcx> {
     }
 }
 
-impl ItemLikeVisitor<'v> for UnsafetyChecker<'tcx> {
+impl<'v, 'tcx> ItemLikeVisitor<'v> for UnsafetyChecker<'tcx> {
```
Comment (jackh726) [inline compiler/rustc_typeck/src/coherence/unsafety.rs:86]: And here

**Review by jackh726 — APPROVED** [review]
A few comments. This needs a fmt, but otherwise it's really just nits.

r=me after fmt at least

**Comment by scottmcm** [comment]
@Patrick-Poitras You can run `x.py fmt` to update the formatting to match what CI is expecting.

File: compiler/rustc_typeck/src/coherence/builtin.rs (hunk 3 above)
Comment (Patrick-Poitras) [reply compiler/rustc_typeck/src/coherence/builtin.rs:111]: I tried to mostly follow the conventions of surrounding functions, but the rest of the file, other than this function and the one who follows use `'tcx'`, so I've changed both to follow the rest of the file.

**Pull request merged**